src/wallet/wallet.cpp

```diff
@@ -1419,6 +1419,38 @@ void CWallet::transactionAddedToMempool(const CTransactionRef& tx) {
     if (it != mapWallet.end()) {
         RefreshMempoolStatus(it->second, chain());
     }
+
+    const Txid& conflict_txid = tx->GetHash();
+    const Wtxid& conflict_wtxid = tx->GetWitnessHash();
+
+    for (const CTxIn& tx_in : tx->vin) {
+        // If the wallet contains no transactions that spend the same prevout, continue
+        if (mapTxSpends.count(tx_in.prevout) == 0) continue;
+
+        std::pair<TxSpends::const_iterator, TxSpends::const_iterator> range;
+        range = mapTxSpends.equal_range(tx_in.prevout);
+
+        // For each wallet transaction spending this prevout...
+        for (TxSpends::const_iterator _it = range.first; _it != range.second; ++_it) {
+            // if this tx is the same as the tx which was just added to the mempool, continue
+            auto it = mapWallet.find(_it->second);
+            if (it->second.tx->GetWitnessHash() == conflict_wtxid) continue;
```

Comment:
> Would be good to explain that you are doing the mapWallet lookup because the wtx could have been updated.

I just realized that the wtx could not have been updated, because for it to be updated, `transactionAddedToMempool` would need to be called, which would be after this call, right? So I have reverted the change which compares witness hashes. 

> Also, as you already know that the output belongs to one of the wallet's txs, could:

Yes, I have implemented this.